core/state_processor.go

```diff
@@ -317,6 +322,10 @@ func ApplyCrossShardDeposit(config *params.ChainConfig, bc ChainContext, header
 
 func PayNativeTokenAsGas(evmState vm.StateDB, config *params.ChainConfig, tokenID, gas uint64,
 	gasPriceInNativeToken *big.Int) (uint8, *big.Int, error) {
+	// not need to check gas(uint64)
+	if tokenID > qkcCmn.TOKENIDMAX || qkcCmn.BiggerThanUint128Max(gasPriceInNativeToken) {
```

Comment:
missing check(gas <= UINT128_MAX) ?

Reply:
gas type is uint64
uint64  small than Uint128Max